### PR TITLE
[CHK-2028] feat(wallet): handling concurrency 

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Use Mongo auditing to trace

- creationDate
- lastUpdateDate
- version

For wallet document

This solution requires to refactor the id of the document, using internal id as a standard id and a WalletId to continue to use UUID. This result could be achieved also editing the standard entity id format

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/wallet/documents/wallets/Wallet.kt [hunk 1]
```diff
@@ -1,38 +1,42 @@
...
+import org.springframework.data.annotation.CreatedDate
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.LastModifiedDate
+import org.springframework.data.annotation.Version
 import org.springframework.data.mongodb.core.mapping.Document
 
 @Document("wallets")
 data class Wallet(
-    val id: String,
+    @PartitionKey val walletId: String,
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/wallet/documents/wallets/Wallet.kt:18]: why use this new field? we could leave id and continue generating it on the application side.

File: src/main/kotlin/it/pagopa/wallet/documents/wallets/Wallet.kt [hunk 2]
```diff
@@ -1,38 +1,42 @@
...
     val status: String,
-    val creationDate: String,
-    val updateDate: String,
     val paymentMethodId: String,
     val paymentInstrumentId: String?,
     val contractId: String?,
     val applications: List<Application>,
-    val details: WalletDetails<*>?
+    val details: WalletDetails<*>?,
+    @CreatedDate val creationDate: Instant? = null,
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/wallet/documents/wallets/Wallet.kt:26]: `creationDate`, `updateDate` and `version` are not optional, there is no valid document/wallet without these three fields, so they must be mandatory and initialized only when creating a new wallet

File: src/main/kotlin/it/pagopa/wallet/domain/wallets/Wallet.kt [hunk 3]
```diff
@@ -37,11 +36,9 @@ import java.time.Instant
  */
 @AggregateRoot
 data class Wallet(
-    @AggregateRootId val id: WalletId,
+    @AggregateRootId val walletId: WalletId,
     val userId: UserId,
     var status: WalletStatusDto,
-    val creationDate: Instant,
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/wallet/domain/wallets/Wallet.kt:43]: these two fields are parts of the aggregate, they cannot be deleted, furthermore we can consider also inserting the version field in the aggregate

**Comment by ciuffagianluca** [comment]
Handled in #68

**Pull request closed without merging**